app/ai_backend/test_genesis_connector.py

```
@@ -789,56 +1674,101 @@ def test_send_request_with_binary_payload(self, mock_post):
+        """Test handling of special characters in headers."""
+        special_headers = {
+            'X-Custom-Header': 'value with spaces',
+            'X-Unicode-Header': 'café',
+            'X-Special-Chars': '!@#$%^&*()',
+            'X-Quotes': 'value with "quotes"',
+            'X-Newlines': 'value\nwith\nnewlines',
+            'X-Tabs': 'value\twith\ttabs',
         }
         
-        result = self.connector.send_request(nested_payload)
+        connector = GenesisConnector(config={
```

Ruff (0.11.9)

Local variable `connector` is assigned to but never used

Remove assignment to unused variable `connector`

(F841)